### ISSUE TITLE
Undetected unreachable code made by NaN

### ISSUE THREAD

**Issue by falsandtru** [microsoft/TypeScript#36414]
The compiler wouldn't make comparison with NaN an error but developers would become aware of that mistake via this unreachable code detection.

**TypeScript Version:**  3.7.x-dev.20200123

**Search Terms:**

**Code**

```ts
declare const a: any;
if (a === NaN) {
    // Unreachable
}
```

**Expected behavior:**
Detect unreachable code.
**Actual behavior:**
Not detected.
**Playground Link:** http://www.typescriptlang.org/play/index.html?target=99&ts=3.8.0-dev.20200123#code/CYUwxgNghgTiAEYD2A7AzgF3lAXNlAngNwBQAlgGbwAUU8AvI-AHJTMCU8A3ifH-AHoB8AKoo4UMAAsoAIwggSAXyA

**Related Issues:** https://github.com/microsoft/TypeScript/issues/28682#issuecomment-552352389 #35835

**Comment by RyanCavanaugh** [comment]
Declining on complexity grounds
